### DISCUSSION TITLE
may I convert result from data source returns to resource?

### DISCUSSION THREAD

**Discussion by superwangnan** [SAP/terraform-provider-btp#792] [Q&A]
Hi team,

Just as title, I want to copy / backup subaccount for redeploy or setup a new subaccount. So, I want to know if there is a convenient way to convert the data source returns to resources directly. So, I don't need to key in every setting.

Thanks

**Comment by vipinvkmenon** [comment]
Hi,

I'm not sure if you can take a backup of subaccount /. resources and setup a new account from the same. The reason is that many of the properties for several resources are generated during the time of creation. For E.G GUIDs etc. Hence you would never be able to recreate the subaccount (or any other resource for that matter) exactly the same (restore from backup)

Another thing that you could try is to use datasource to pull the details of a given resource (for eg subaccount) and map it against a resource using the `required` and `optional` fields.  However, I believe you'd need to write some logic to do this activity as its not something that come out of the box from terraform or provider.
